### PR TITLE
[Debt] Replace .js with .ts

### LINKED ISSUE #10934: ♻️ Replace javascript with typescript
## ♻️ Refactor

There are [some files](https://github.com/search?q=repo%3AGCTC-NTGC%2Fgc-digital-talent++language%3AJavaScript&type=code) that are javascript and could be typescript, which is what the codebase uses most of the time.

## 🙋‍♀️ Proposed Solution

Replace javascript with typescript where possible. For example, `apps/web/.storybook/preview.js` becomes `apps/web/.storybook/preview.ts`

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the debt was addressed and expected functionality has not been affected.

- [ ] javascript is replaced with typescript where possible

### PR DESCRIPTION
**Pull request by yonikid15**
🤖 Resolves #10934 

## 👋 Introduction

- Replaces all .js files with .ts where possible.
- Couldn't replace the `jest-preset.js` files. 
  - "A preset should point to an npm module that has a jest-preset.json, jest-preset.js, jest-preset.cjs or jest-preset.mjs file at the root." - [Jest config docs](https://jestjs.io/docs/configuration#preset-string)
- Changing the `package/eslint-config-custom` .js files caused eslint to break. I'm not sure it can recognise the index.ts file as on entry point. However, I'll keep testing it while this is up.

## 🧪 Testing

1. Ensure all jest tests still work
2. Ensure eslint still works correctly
3. Ensure app doesn't have any problems

### CODE REVIEW

**Comment by mnigh** [comment]
@yonikid15 Any reason why `apps/web/.storybook/preview.js` couldn't be changed to `apps/web/.storybook/preview.ts`?

**Comment by yonikid15** [comment]
> @yonikid15 Any reason why `apps/web/.storybook/preview.js` couldn't be changed to `apps/web/.storybook/preview.ts`?

Other than missing it. no :laughing: 
Changes in [ff4b846](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11142/commits/ff4b84629ad14263d3fb2076099834356b09d30e)

File: apps/web/package.json [hunk 1]
```diff
@@ -97,6 +97,7 @@
     "jest-fail-on-console": "^3.3.0",
     "prettier": "^3.3.3",
     "ts-jest": "^29.2.3",
+    "ts-node": "^10.9.2",
```
Comment (mnigh) [inline apps/web/package.json:100]: I don't see `ts-node` imported anywhere: what is it used by and what for?

**Review by mnigh — COMMENTED** [review]
Storybook still works and so does Jest. One question about `ts-node` but otherwise, looks good.

File: apps/web/package.json (hunk 1 above)
Comment (yonikid15) [reply apps/web/package.json:100]: Oh ya sorry,  this is required to read typescript config files.

"To read TypeScript configuration files Jest requires [ts-node](https://npmjs.com/package/ts-node). Make sure it is installed in your project." - [Jest Docs](https://jestjs.io/docs/configuration)

**Review by mnigh — APPROVED** [review]
LGTM

**Pull request merged**